jwst/datamodels/vegamag_offset.py

```diff
@@ -0,0 +1,85 @@
+from .reference import ReferenceFileModel
+
+__all__ = ['FgsImgVegaMagOffsetModel', 'MirImgVegaMagOffsetModel',
+           'NrcImgVegaMagOffsetModel', 'NisImgVegaMagOffsetModel']
+
+
+class FgsImgVegaMagOffsetModel(ReferenceFileModel):
+    """
+    A FGS imaging data model for reference files containing Vega
+    magnitude offsets from AB magnitudes.
+
+    Parameters
+    ----------
+    vegamag_table : numpy table
+        Vega magnitude offsets tables.
+        A table-like object containing row selection criteria made up of
+        instrument mode parameters and aperture correction factors
+        associated with those modes.
+
+        - vegamag_offset: float32
+    """
+
+    schema_url = "http://stsci.edu/schemas/jwst_datamodel/fgsimg_vegamag_offset.schema"
+
+
+class MirImgVegaMagOffsetModel(ReferenceFileModel):
+    """
+    A MIRI imaging data model for reference files containing Vega
+    magnitude offsets from AB magnitudes.
+
+    Parameters
+    ----------
+    vegamag_table : numpy table
```

Comment:
What is the expected parameter type?

Reply:
The model classes should create a valid data model when a file name is passed. Could you add a test?

Reply:
Done.